workbench-core/environments-ui/src/pages/environments/new.tsx

```diff
@@ -0,0 +1,337 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useProjects } from '@aws/workbench-core-accounts-ui';
+import { useDatasets } from '@aws/workbench-core-datasets-ui';
+import { nameRegex, BaseLayout } from '@aws/workbench-core-swb-common-ui';
+import {
+  Box,
+  BreadcrumbGroupProps,
+  SpaceBetween,
+  Form,
+  Header,
+  ExpandableSection,
+  Input,
+  FormField,
+  Button,
+  Container,
+  Link,
+  Textarea,
+  Select,
+  Multiselect
+} from '@cloudscape-design/components';
+import type { NextPage } from 'next';
+import { useRouter } from 'next/router';
+import React, { useEffect, useState } from 'react';
+import { createEnvironment } from '../../api/environments';
+import { useEnvTypeConfigs } from '../../api/environmentTypeConfigs';
+import { useEnvironmentType } from '../../api/environmentTypes';
+import EnvTypeCards from '../../components/EnvTypeCards';
+import EnvTypeConfigCards from '../../components/EnvTypeConfigCards';
+import { CreateEnvironmentForm, CreateEnvironmentFormValidation } from '../../models/Environment';
+import { EnvTypeItem } from '../../models/EnvironmentType';
+import { EnvTypeConfigItem } from '../../models/EnvironmentTypeConfig';
+
+const NewEnvironmentPage: NextPage = () => {
+  // App settings constant
+
+  const router = useRouter();
+  const [isSubmitLoading, setIsSubmitLoading] = useState(false);
+  const [disableSubmit, setDisableSubmit] = useState(true);
+  const [selectedEnvType, setSelectedEnvType] = useState<EnvTypeItem>();
+  const [error, setError] = useState('');
+  const [formData, setFormData] = useState<CreateEnvironmentForm>({});
+  const [formErrors, setFormErrors] = useState<CreateEnvironmentFormValidation>({});
+  const { envTypes, areEnvTypesLoading } = useEnvironmentType();
+  const { envTypeConfigs, areEnvTypeConfigsLoading } = useEnvTypeConfigs(formData?.envTypeId || '');
+  const { projects, areProjectsLoading } = useProjects();
+  const { datasets, areDatasetsLoading } = useDatasets();
+
+  const breadcrumbs: BreadcrumbGroupProps.Item[] = [
+    {
+      text: 'Service Workbench',
+      href: '/'
+    },
+    {
+      text: 'Workspaces',
+      href: '/environments'
+    },
+    {
+      text: 'Create Workspace',
+      href: '/environments/new'
```

Comment:
If this is going to be a common component, these fields should be configurable

Reply:
Another package would be required to handle customizable strings, which is out of scope of this PR. We can work on that once more than one solution depends on this package.